tools/Cargo.toml

```diff
@@ -9,7 +9,8 @@ edition = "2021"
 chewing = { version = "0.5.1-alpha.1", path = ".." }
 thiserror = "1.0.0"
 anyhow = "1.0.0"
-clap = "4.1.0"
+clap = "~4.0.0"
+clap_lex = "=0.3.0"
```

Comment:
Thank you for taking care of MSRV! And sorry for bombarding your pull request again.

For dependency requirements other than the default caret requirements, I would suggest adding comments explaining why we need these special cases, and also a TODO about when to remove these bounds.

Reply:
`argh` does look great and concise. However, I mostly use clap throughout my Rust journey,  not really familiar with other alternatives, sorry 😞. If you just want to derive arguments from structs, clap can achieve that since clap v3 IIRC. <https://docs.rs/clap/latest/clap/_derive/index.html>